[Repository] Fix validation failing when deleting a quickstart

## Relates to

https://github.com/newrelic/newrelic-quickstarts/issues/668

## Summary
Pointed out as part of [this PR](https://github.com/newrelic/newrelic-quickstarts/pull/744) that some of the validation checks failed when they shouldn't. This occurred on the deletion of a quickstart from the repo.

When a PR involves deletion of a quickstart, we should not validate anything (at the moment). Because we are attempting to read a deleted file, one of the checks is failing.

The exact error occurs in the `Validate new files` step and is:
```
$ node validate-quickstart-install-plans.js https://api.github.com/repos/newrelic/newrelic-quickstarts/pulls/744/files
node:fs:585
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/quickstarts/pivotal-tracker/config.yml'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at readYamlFile (/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/helpers.js:14:19)
    at readQuickstartFile (/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/helpers.js:38:7)
    at /home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/validate-quickstart-install-plans.js:37:30
    at Array.map (<anonymous>)
    at getConfigInstallPlans (/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/validate-quickstart-install-plans.js:35:22)
    at validateInstallPlanIds (/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/validate-quickstart-install-plans.js:67:35)
    at main (/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/utils/validate-quickstart-install-plans.js:99:3)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/runner/work/newrelic-quickstarts/newrelic-quickstarts/quickstarts/pivotal-tracker/config.yml'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

## Acceptance Criteria
- [ ] Validate install plan ids job doesn't fail when PR involves deleting a quickstart.

--- a/utils/__tests__/validate_install_quickstart_plans.test.js
+++ b/utils/__tests__/validate_install_quickstart_plans.test.js
@@ -46,7 +46,7 @@ describe('Action: validate install plan id', () => {
     expect(global.console.error).not.toHaveBeenCalled();
   });
 
-  test.only(`succeeds when valid quickstart doesn't contain any install plan`, () => {
+  test(`succeeds when valid quickstart doesn't contain any install plan`, () => {
     const files = mockGithubAPIFiles([validQuickstartWithoutInstallPlan]);
     githubHelpers.filterQuickstartConfigFiles.mockReturnValueOnce(files);
 
@@ -80,5 +80,15 @@ describe('Action: validate install plan id', () => {
 
     validateInstallPlanIds(files);
     expect(global.console.error).toHaveBeenCalledTimes(5);
+  });
+
+  test('does not fail for deleted quickstart', () => {
+    const removedQuickstartFilename = 'fake-removed-quickstart/config.yml';
+    const files = mockGithubAPIFiles([removedQuickstartFilename]);
+    files[0].status = 'removed';
+    githubHelpers.filterQuickstartConfigFiles.mockReturnValueOnce(files);
+
+    validateInstallPlanIds(files);
+    expect(global.console.error).toHaveBeenCalledTimes(0);
   });
 });

--- a/utils/validate-quickstart-install-plans.js
+++ b/utils/validate-quickstart-install-plans.js
@@ -64,8 +64,11 @@ const getInstallPlansNoMatches = (configInstallPlanFiles, installPlanIds) => {
  */
 const validateInstallPlanIds = (githubFiles) => {
   const configFiles = filterQuickstartConfigFiles(githubFiles);
+  const existingConfigFiles = configFiles.filter(
+    (cf) => cf.status !== 'removed'
+  ); // Filter out deleted files
 
-  const configInstallPlansFiles = getConfigInstallPlans(configFiles);
+  const configInstallPlansFiles = getConfigInstallPlans(existingConfigFiles);
 
   const installPlanIds = getAllInstallPlanIds();
 

